slowmovie.py

```diff
@@ -10,205 +10,349 @@
-        currentVideo = args.file
-    else: 
-        print ('%s not found' %args.file)
-
-print("The current video is %s" %currentVideo)
-
-# Ensure this is the correct driver for your particular screen 
-epd = epd7in5_V2.EPD()
-
-# Initialise and clear the screen 
-epd.init()
-epd.Clear()    
+# Add hooks for interrupt signal
+signal.signal(signal.SIGTERM, signal_handler)
+signal.signal(signal.SIGINT, signal_handler)
 
-currentPosition = 0
+# Move to the directory where this code is
+os.chdir(os.path.dirname(os.path.realpath(__file__)))
 
-# Open the log file and update the current position 
+# Name of videos directory and logs directory. Videos directory is specified by CLI --directory
+viddir = args.directory
+logdir = 'logs'
 
-log = open(logdir + '%s<progress'%currentVideo)
-for line in log:
-    currentPosition = float(line)
+# Create logs directory if missing
+if not os.path.isdir(logdir):
+    os.mkdir(logdir)
 
-if args.start:
-    print('Start at frame %f' %float(args.start))
-    currentPosition = float(args.start)
+if args.random_frames:
+    print('In random mode')
+else:
+    print ('In play-through mode')
 
-# Ensure this matches your particular screen 
-width = 800 
-height = 480 
+# Pick which video to play
 
-inputVid = viddir + currentVideo
+# First, try the --file CLI argument...
+currentVideo = args.file
 
-# Check how many frames are in the movie 
-frameCount = int(ffmpeg.probe(inputVid)['streams'][0]['nb_frames'])
-print("there are %d frames in this video" %frameCount)
-
-while 1: 
-
-    if args.random:
-        frame = random.randint(0,frameCount)
-    else: 
-        frame = currentPosition
-
-    msTimecode = "%dms"%(frame*41.666666)
-        
-    # Use ffmpeg to extract a frame from the movie, crop it, letterbox it and save it as grab.jpg 
-    generate_frame(inputVid, 'grab.jpg', msTimecode, width, height)
-    
-    # Open grab.jpg in PIL  
-    pil_im = Image.open("grab.jpg")
-    
-    # Dither the image into a 1 bit bitmap (Just zeros and ones)
-    pil_im = pil_im.convert(mode='1',dither=Image.FLOYDSTEINBERG)
+# ...then try a random video, if --random-file was selected...
+if not currentVideo and args.random_file:
+    videos = list(filter(is_vid, os.listdir(viddir)))
+    if videos:
+        currentVideo = random.choice(videos)
 
-    # display the image 
-    epd.display(epd.getbuffer(pil_im))
-    print('Diplaying frame %d of %s' %(frame,currentVideo))
-    
-    currentPosition = currentPosition + increment 
-    if currentPosition >= frameCount:
-        currentPosition = 0
-        log = open(logdir + '%s<progress'%currentVideo, 'w')
-        log.write(str(currentPosition))
-        log.close() 
-    
-        thisVideo = movieList.index(currentVideo)
-        if thisVideo < len(movieList)-1:
-            currentVideo = movieList[thisVideo+1]
+# ...then try the nowPlaying file, which stores the currently-playing video...
+if not currentVideo and os.path.isfile('nowPlaying'):
```

Comment:
Based on the ordering of this the --file flag __always__ trumps the ```nowPlaying``` file? Is this problematic. Every time you restart the program, or the entire OS, you'll restart the video instead of loading the last played frame? Would it be better to  check ```nowPlaying``` first? Trying to think of a way to resume on subsequent runs better.

Reply:
The current position is stored in per-video files in the logs folder. It has nothing to do with the --file argument or the nowPlaying file.

Reply:
This is a great line of discussion, thanks for emphasizing it, Rob. The way I imagine this being used is that I start it up with something like `slowmovie.py -f 'spiritedaway.mkv'` to pick what movie to start with. But then I set the pi every reboot to run `slowmovie.py` without any file options. This way when I reboot, it'll prioritize the nowPlaying file if that's changed because I finished a movie.

Notice that's also what's instructed in [Tom's original post](https://debugger.medium.com/how-to-build-a-very-slow-movie-player-in-2020-c5745052e4e4):
```
cd SlowMoviesudo
python slowmovie.py
```

This order works with that usage.